### DISCUSSION TITLE
require-ing MathJax fully working in Observerable notebook, but NOT in Quarto ojs cell

### DISCUSSION THREAD

**Discussion by stuvjordan-uroc** [quarto-dev/quarto-cli#1682] [Q&A]
I'm wanting to include some Observable charts in a website I'm building with Quarto.  I'd like these charts to include nicely-typeset math, and for that purpose I'm wanting to use MathJax's `tex2svg` function in the source code.  Relative to other math typesetting solutions one can use in Observable, `tex2svg` is nice because it outputs pure SVG paths...Thus they scale as the svg containing them is resized without any additional code, so the math looks great on every screen size.

I'm running into this **very** weird difference in the result of a call to `require` when that call is made in an Observable Notebook, as opposed to when I make it in an `ojs` cell in a Quarto sourcefile.

First, let's look at what happens when we require `tex2svg` in an Observable Notebook.  You can see the results in [this notebook](https://observablehq.com/@stuvjordan/requiring-mathjax-tex2svg).  Open up the object in the first cell after the title and notice the full list of methods that the object returned by the call to `require` includes.  Notice the list includes the method `.tex2svg`, which is the function I need.  Below is a screenshot.  You'll see `tex2svg` as the fourth method from the bottom.

![image](https://user-images.githubusercontent.com/78857503/182877558-aec76537-8983-4f36-876a-8053401248f8.png)

OK, great, so we can use `tex2svg` in an Observable notebook!

But what happens when we do the same call to `require` in an `ojs` cell in a `.qmd` file?  Here are two cells from my current `.qmd` source:
```{ojs}
//| output: true
MathJax = require('https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-svg.js').catch(
  () => window['MathJax']
)
```

```{ojs}
MathJax
```
Here's a screenshot of the output from the second of the above two `ojs` cells when I render/preview the sourcefile:

![image](https://user-images.githubusercontent.com/78857503/182878897-538a83da-691c-47be-b621-d7935ada8515.png)

**Notice that the list of methods does not include `tex2svg`!!!!**  As a result, when I do something like
```{ojs}
MathJax.tex2svg(
    String.raw`\int_0^1 \exp(x)dx`
 );
```
In the quarto sourcefile, the output on render/preview is:

![image](https://user-images.githubusercontent.com/78857503/182879785-70a332fe-3e53-488f-8d99-a671d42624b7.png)

I'm kind of amazed by this.

Help?

**Comment by stuvjordan-uroc** [comment]
I've discovered the problem.

In a `qmd` file that includes math in the main body, as in `here is some math in a sentence $\sum_n \frac{1}{2}$`, `require`-ing MathJax in an `ojs` cell results in `tex2svg` not being included in the objected returned from `require`.

On the other hand, in a `qmd` file that does not include math in the main body, the `require` call in an `ojs` cell performs as expected.

Should I report this as a bug?  Or is this expected behavior?

**Reply by kwangkim** [reply]
A `quarto template` is using `chtml` instead of `svg` as a default.
<https://github.com/quarto-dev/quarto-cli/blob/8d0e6ba7e641abac0e2b931aebaa4ef706e5c13b/src/command/render/template.ts#L189>

I found two possible temporary solutions.
- You may use `katex` as the default `html-math-method` for `quarto`. 
<https://quarto.org/docs/reference/formats/html.html>

-  Use `MJ`  in <https://observablehq.com/@kkim/mj> which is modified from <https://observablehq.com/@andy0130tw/mathjax-v3>.
Note: `load: ['output/svg']` may not be necessary. I do not know how it works.

To solve this problem permanently, I think we can use one of ideas  to fix the `quarto` template.
- <https://github.com/mathjax/MathJax/issues/2705#issuecomment-859742446>
- <http://docs.mathjax.org/en/latest/web/webpack.html#custom-component>
